### DISCUSSION TITLE
API returns a file to download

### DISCUSSION THREAD

**Discussion by ionv06** [sqlpage/SQLPage#604] [Q&A] [answered]
Hi,
I need to consume an API whose response is a .zip file. I tried to use sqlpage.fetch but I get the error 'Unable to convert the response from https://api.myapiad
dress.com/?id=xxxxx to a string. Only UTF-8 responses are supported.' Considering that after downloading I should unzip the .zip file (which contains an xml that needs to be processed), what would be the best option? Can I solve it only with SQLPage (but without sqlpage.exec) or should I implement an API in NodeJS or something else, and have that API download the zip and unzip) and call that API from SQLPAge?

**Comment by lovasoa** [comment, marked as answer]
Hi! 
  The best option is probably to use sqlpage.exec, and write a short shell script that downloads your file and unzips it. 
  https://sql.datapage.app/functions.sql?function=exec#function
